validator/build.py

```diff
@@ -194,30 +194,58 @@ def GenerateValidateBin(out_dir):
   f.write("""
       var fs = require('fs');
       var path = require('path');
+      var http = require('http');
+      var https = require('https');
+      var url = require('url');
 
       function main() {
         if (process.argv.length < 3) {
-          console.error('usage: validate <file.html>');
+          console.error('usage: validate <file.html or url>');
           process.exit(1)
         }
         var args = process.argv.slice(2);
         var full_path = args[0];
-        var filename = path.basename(full_path);
-        var contents = fs.readFileSync(full_path, 'utf8');
-        var results = amp.validator.validateString(contents);
-        var output = amp.validator.renderValidationResult(results, filename);
-
-        if (output[0] === 'PASS') {
-          for (var i = 0; i < output.length; ++i) {
-            console.info(output[i]);
+        var validateFile = function(contents, filename) {
```

Comment:
Can you make this its own function please? As in, just like main is a function, have function validateFile? I think it'll be easier to read that way, less nesting.